lib/app-dependencies.js

```diff
@@ -94,62 +121,82 @@ async function addSkyPeerDependencies(dependencies) {
 
 async function upgradeDependencies(dependencies) {
   if (dependencies) {
-    const dependencyPromises = [];
-
-    // Make sure any versions listed as 'latest' are converted to
-    // the numeric equivalent.
-    await setDependencyVersions(dependencies);
 
-    const packageNames = Object.keys(dependencies);
+    const promises = [];
 
-    for (const packageName of packageNames) {
+    for (const packageName in dependencies) {
       const currentVersion = dependencies[packageName];
 
-      let dependencyPromise;
-
-      if (/^[0-9]+\./.test(currentVersion)) {
-        let majorVersion = currentVersion.split('.')[0];
-
-        // Handle prerelease versions.
-        if (/-(rc|alpha|beta)\./.test(currentVersion)) {
-          majorVersion = `^${currentVersion}`;
-        }
-
-        logger.info(`Getting latest ${packageName} version for major version ${majorVersion}...`);
-
-        dependencyPromise = latestVersion(
-          packageName,
-          {
-            version: `${majorVersion}`
-          }
-        );
-      } else {
-        dependencyPromise = Promise.resolve(currentVersion);
-      }
-
-      dependencyPromises.push(dependencyPromise);
-    }
+      let versionRange;
 
-    const versionNumbers = await Promise.all(dependencyPromises);
-
-    packageNames.forEach((packageName, index) => {
-      const versionNumber = versionNumbers[index];
       switch (packageName) {
         case 'typescript':
-          logger.info('Skipping TypeScript because it does not use semantic versioning.');
+          versionRange = '~3.6.4';
+          logger.info(
+            `Note: Using version range ${packageName}@${versionRange} because TypeScript does not support semantic versioning.`
+          );
           break;
         case 'zone.js':
-          logger.info('Skipping Zone.js because Angular requires a specific minor version.');
+          versionRange = '~0.10.2';
+          logger.info(
+            `Note: Using version range ${packageName}@${versionRange} because Angular requires a specific minor version of ${packageName}.`
+          );
+          break;
+        case 'ts-node':
+          versionRange = '~8.6.0';
+          logger.info(
+            `Note: Using version range ${packageName}@${versionRange} because Angular requires a specific minor version of ${packageName}.`
+          );
           break;
         default:
-          if (dependencies[packageName] === versionNumber) {
-            logger.info(`Package ${packageName} already on latest version (${versionNumber})`);
+          if (
+            /^[0-9]+\./.test(currentVersion) ||
+            /-(rc|alpha|beta)\./.test(currentVersion)
+          ) {
+            versionRange = `^${currentVersion.replace('^', '').replace('~', '')}`;
```

Comment:
Why is this necessary?

Reply:
Great catch; I wasn't taking into consideration how many formats a version range can take. Per your suggestion, I'm now using the `semver` package.